### PR TITLE
Add EKS Istio Dashboards

### PR DESCRIPTION
**Pull request by dms486**
### What does this PR do?

Adding the EKS Istio dashboards example

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted. Consult the CONTRIBUTING guide for submitting pull-requests.

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples) to support my PR (when applicable)
- [ ] Yes, I have updated the [Pages](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/docs) for this feature

**Note**: Not all the PRs required examples and docs.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by bonclay7 — COMMENTED** [review]
Can you guys install https://pre-commit.com/ and run `pre-commit run -a` ?

**Comment by dms486** [comment]
@bonclay7 sure, this is complete

File: docs/eks/istio.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+```hcl
+
+module "eks_monitoring" {
+   ...
+   enable_istio = true
+}
+```
+
+You can further customize the NGINX pattern by providing `nginx_config` [options](https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/README.md#input_nginx_config).
```
Comment (bonclay7) [inline docs/eks/istio.md:23]: Can you replace nginx here with Istio?

Comment (dms486) [reply docs/eks/istio.md]: @bonclay7 sure, it's updated now

**Comment by bonclay7** [comment]
thanks let me re-run the tests

**Review by bonclay7 — COMMENTED** [review]
Nice job! few minor changes, and we'll be able to ship

File: docs/eks/istio.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+### 3. Explore the Istio dashboards
+
+Log back into your Amazon Managed Grafana workspace and navigate to the dashboard side panel. Click on the `Observability Accelerator Dashboards` folder and open the `Istio Service` Dashboard. Use the Service dropdown menu to select the `reviews.default.svc.cluster.local` service. This gives details about metrics for the service, client workloads (workloads that are calling this service), and service workloads (workloads that are providing this service).
+
+Explore the Istio Control Plane, Mesh, and Performance dashboards as well.
+
+## Destroy
+
+To teardown and remove the resources created in this example:
```
Comment (bonclay7) [inline docs/eks/istio.md:175]: Can we add kubectl commands to delete resources of the istio sample app too?

```
kubectl delete -f samples/bookinfo/platform/kube/bookinfo.yaml
kubectl delete -f samples/bookinfo/networking/bookinfo-gateway.yaml
```

File: docs/eks/istio.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+## Deploy
+
+Simply run this command to deploy.
+
+```bash
+terraform apply
+```
+
+!!! note
+    To see the complete ISTIO example, open the [example on the repository](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples/eks-istio)
```
[the remark this answers is not in the document]
Comment (dms486) [reply docs/eks/istio.md]: Yes, I consolidated these

File: docs/eks/istio.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    To setup your alert receiver, with Amazon SNS, follow [this documentation](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-alertmanager-receiver.html)
+
+## Deploy an example application to visualize metrics
+
+In this section we will deploy sample application and extract metrics using AWS OpenTelemetry collector
+
+### 1. Deploy the Bookinfo Application
+
+```sh
+#https://istio.io/latest/docs/examples/bookinfo/
```
[the remark this answers is not in the document]
Comment (dms486) [reply docs/eks/istio.md]: Yes, I added instructions for the Istio Bookinfo app

File: docs/eks/istio.md (hunk 2 above)
Comment (dms486) [reply docs/eks/istio.md:168]: Thanks, these were added

File: docs/eks/istio.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+```
+6. Validate that there are no issues with the Istio configuration
+```sh
+istioctl analyze
+```
+7. Get the DNS name of the load balancer for the Istio gateway
+```sh
+GATEWAY_URL=$(kubectl get svc istio-ingressgateway -n istio-system -o=jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+```
+Additional details can be found on Istio's [Getting Started](https://istio.io/latest/docs/setup/getting-started/) documentation
```
[the remark this answers is not in the document]
Comment (dms486) [reply docs/eks/istio.md]: Thanks, I moved this up to the description

File: docs/eks/istio.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+Open the Amazon Managed Service for Prometheus console and view the details of your workspace. Under the `Rules management` tab, you will find new rules deployed.
+
+<img width="1054" alt="image" src="https://user-images.githubusercontent.com/47993564/236844084-80c754e3-4fe1-45bb-8361-181432675469.jpeg">
+
+!!! note
+    To setup your alert receiver, with Amazon SNS, follow [this documentation](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-alertmanager-receiver.html)
+
+## Deploy an example application to visualize metrics
+
+In this section we will deploy a sample application and extract metrics using the AWS OpenTelemetry collector
```
[the remark this answers is not in the document]
Comment (dms486) [reply docs/eks/istio.md]: Thanks, I added some additional instructions. Let me know if I can add anything else

File: docs/images/istio-dashboards.png
Comment (bonclay7) [inline docs/images/istio-dashboards.png:1]: Can we use the Github generated links? it renders better and contributes to making the repo lighter for clones

**Review by bonclay7 — COMMENTED** [review]
@dms486 thanks for your patience on this. Please address my new comments, run pre-commit and we'll get this merged today

File: docs/eks/istio.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+1. Cross-region Amazon Managed Prometheus workspace
+
+If your existing Amazon Managed Prometheus workspace is in another AWS Region,
+add this `managed_prometheus_region=xxx` and `managed_prometheus_workspace_id=ws-xxx`.
+
+2. Cross-region Amazon Managed Grafana workspace
+
+If your existing Amazon Managed Prometheus workspace is in another AWS Region,
+add this `managed_prometheus_region=xxx` and `managed_prometheus_workspace_id=ws-xxx`.
```
[the remark this answers is not in the document]
Comment (dms486) [reply docs/eks/istio.md]: Sure, this has been removed

File: docs/images/istio-dashboards.png
Comment (dms486) [reply docs/images/istio-dashboards.png:1]: I made the switch here, let me know if that doesn't look right. I ran the pre-commit locally too

**Review by bonclay7 — APPROVED** [review]
Thanks a LOT!

**Pull request merged**